src/node/cli.ts

```diff
@@ -626,7 +630,20 @@ function bindAddrFromAllSources(...argsConfig: Args[]): Addr {
 }
 
 export const shouldRunVsCodeCli = (args: Args): boolean => {
-  return !!args["list-extensions"] || !!args["install-extension"] || !!args["uninstall-extension"]
+  // Create new interface with only these keys
+  // Pick<Args, "list-extensions" | "install-extension" | "uninstall-extension">
+  // Get the keys of new interface
+  // keyof ...
+  // Turn that into an array
+  // Array<...>
+  type ExtensionArgs = Array<keyof Pick<Args, "list-extensions" | "install-extension" | "uninstall-extension">>
```

Comment:
I like the overall strategy to make this type-safe and get the benefit from the compiler! The only thing I find a bit awkward is that we have to reference the `"list-extensions" | "install-extensions" | "uninstall-extensions"` twice here.

I wonder if we could get most of the benefits, without having to define these multiple times, by doing:

```
type ArgumentKeys = Array<keyof Args>;
const extensionRelatedArgs: ArgumentKeys = ["list-extensions", "install-extension", "uninstall-extension"];
```

That way - the compiler will still flag an error if these get out-of-sync, but you only have to define them once here in `shouldRunVsCodeCli`

Reply:
Oh this is a huge improvement. Waaaaay better. Didn't think about that. Thanks for providing this suggestion! I'll incorporate this suggestion before merging! 🙌